### ISSUE TITLE
`map_agg` returns different results under fuzzer

### ISSUE THREAD

**Issue by Yuhta** [facebookincubator/velox#3418]
### Description

https://app.circleci.com/pipelines/github/facebookincubator/velox/17199/workflows/1a27595b-09a1-4ec6-a4c0-6740a7e21220/jobs/107513

### Error Reproduction

Run aggregation fuzzer with seed 142768294 and version 1299c919872774e2558c30f4fc292d6772944bca

```
#!/bin/bash -eo pipefail
echo -e "The logs will be saved to \"/tmp/aggregate_fuzzer.log\" and uploaded" \
        "as a job artifact (check the \"Artifacts\" tab above)."
_build/debug/velox/exec/tests/velox_aggregation_fuzzer_test \
    --seed ${RANDOM} \
    --duration_sec 3600 \
    --logtostderr=1 \
    --minloglevel=0 \
  2>&1 | tee /tmp/aggregate_fuzzer.log || ( \
    tail -n 1000 /tmp/aggregate_fuzzer.log;
    echo "FAIL: Aggregate Fuzzer run failed";
    exit 1; \
  )
echo -e "\n\nAggregate Fuzzer run finished successfully."
```

### Relevant logs

```bash
I1205 03:38:27.850936  7359 AggregationFuzzer.cpp:481] ==============================> Started iteration 2177 (seed: 142768294)
I1205 03:38:27.853351  7359 AggregationFuzzer.cpp:561] Executing query plan: 
-- Project[expressions: (g0:BOOLEAN, ROW["g0"]), (p1:ARRAY<SMALLINT>, array_sort(map_keys(ROW["a0"])))] -> g0:BOOLEAN, p1:ARRAY<SMALLINT>
  -- Aggregation[SINGLE [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
    -- Values[1000 rows in 10 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
I1205 03:38:27.856894  7359 AggregationFuzzer.cpp:576] [ROW ROW<g0:BOOLEAN,p1:ARRAY<SMALLINT>>: 3 elements, no nulls]
I1205 03:38:27.878767  7359 AggregationFuzzer.cpp:170] Testing plan #0
I1205 03:38:27.878787  7359 AggregationFuzzer.cpp:561] Executing query plan: 
-- Project[expressions: (g0:BOOLEAN, ROW["g0"]), (p1:ARRAY<SMALLINT>, array_sort(map_keys(ROW["a0"])))] -> g0:BOOLEAN, p1:ARRAY<SMALLINT>
  -- Aggregation[FINAL [g0] a0 := map_agg("a0")] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
    -- Aggregation[PARTIAL [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
      -- Values[1000 rows in 10 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
I1205 03:38:27.882959  7359 AggregationFuzzer.cpp:576] [ROW ROW<g0:BOOLEAN,p1:ARRAY<SMALLINT>>: 3 elements, no nulls]
I1205 03:38:27.883507  7359 AggregationFuzzer.cpp:173] Testing plan #0 with spilling
I1205 03:38:27.883513  7359 AggregationFuzzer.cpp:561] Executing query plan: 
-- Project[expressions: (g0:BOOLEAN, ROW["g0"]), (p1:ARRAY<SMALLINT>, array_sort(map_keys(ROW["a0"])))] -> g0:BOOLEAN, p1:ARRAY<SMALLINT>
  -- Aggregation[FINAL [g0] a0 := map_agg("a0")] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
    -- Aggregation[PARTIAL [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
      -- Values[1000 rows in 10 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
I1205 03:38:27.887465  7359 AggregationFuzzer.cpp:576] [ROW ROW<g0:BOOLEAN,p1:ARRAY<SMALLINT>>: 3 elements, no nulls]
I1205 03:38:27.888191  7359 AggregationFuzzer.cpp:170] Testing plan #1
I1205 03:38:27.888201  7359 AggregationFuzzer.cpp:561] Executing query plan: 
-- Project[expressions: (g0:BOOLEAN, ROW["g0"]), (p1:ARRAY<SMALLINT>, array_sort(map_keys(ROW["a0"])))] -> g0:BOOLEAN, p1:ARRAY<SMALLINT>
  -- Aggregation[FINAL [g0] a0 := map_agg("a0")] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
    -- Aggregation[INTERMEDIATE [g0] a0 := map_agg("a0")] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
      -- Aggregation[PARTIAL [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
        -- Values[1000 rows in 10 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
I1205 03:38:27.892874  7359 AggregationFuzzer.cpp:576] [ROW ROW<g0:BOOLEAN,p1:ARRAY<SMALLINT>>: 3 elements, no nulls]
I1205 03:38:27.893404  7359 AggregationFuzzer.cpp:173] Testing plan #1 with spilling
I1205 03:38:27.893411  7359 AggregationFuzzer.cpp:561] Executing query plan: 
-- Project[expressions: (g0:BOOLEAN, ROW["g0"]), (p1:ARRAY<SMALLINT>, array_sort(map_keys(ROW["a0"])))] -> g0:BOOLEAN, p1:ARRAY<SMALLINT>
  -- Aggregation[FINAL [g0] a0 := map_agg("a0")] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
    -- Aggregation[INTERMEDIATE [g0] a0 := map_agg("a0")] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
      -- Aggregation[PARTIAL [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
        -- Values[1000 rows in 10 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
I1205 03:38:27.898073  7359 AggregationFuzzer.cpp:576] [ROW ROW<g0:BOOLEAN,p1:ARRAY<SMALLINT>>: 3 elements, no nulls]
I1205 03:38:27.898788  7359 AggregationFuzzer.cpp:170] Testing plan #2
I1205 03:38:27.898798  7359 AggregationFuzzer.cpp:561] Executing query plan: 
-- Project[expressions: (g0:BOOLEAN, ROW["g0"]), (p1:ARRAY<SMALLINT>, array_sort(map_keys(ROW["a0"])))] -> g0:BOOLEAN, p1:ARRAY<SMALLINT>
  -- Aggregation[FINAL [g0] a0 := map_agg("a0")] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
    -- LocalPartition[REPARTITION] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
      -- Aggregation[PARTIAL [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
        -- Values[300 rows in 3 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
      -- Aggregation[PARTIAL [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
        -- Values[300 rows in 3 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
      -- Aggregation[PARTIAL [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
        -- Values[200 rows in 2 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
      -- Aggregation[PARTIAL [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
        -- Values[200 rows in 2 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
I1205 03:38:27.902276  7359 AggregationFuzzer.cpp:576] [ROW ROW<g0:BOOLEAN,p1:ARRAY<SMALLINT>>: 3 elements, no nulls]
I1205 03:38:27.902827  7359 AggregationFuzzer.cpp:173] Testing plan #2 with spilling
I1205 03:38:27.902834  7359 AggregationFuzzer.cpp:561] Executing query plan: 
-- Project[expressions: (g0:BOOLEAN, ROW["g0"]), (p1:ARRAY<SMALLINT>, array_sort(map_keys(ROW["a0"])))] -> g0:BOOLEAN, p1:ARRAY<SMALLINT>
  -- Aggregation[FINAL [g0] a0 := map_agg("a0")] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
    -- LocalPartition[REPARTITION] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
      -- Aggregation[PARTIAL [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
        -- Values[300 rows in 3 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
      -- Aggregation[PARTIAL [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
        -- Values[300 rows in 3 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
      -- Aggregation[PARTIAL [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
        -- Values[200 rows in 2 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
      -- Aggregation[PARTIAL [g0] a0 := map_agg(ROW["c0"],ROW["c1"])] -> g0:BOOLEAN, a0:MAP<SMALLINT,TIMESTAMP>
        -- Values[200 rows in 2 vectors] -> c0:SMALLINT, c1:TIMESTAMP, g0:BOOLEAN
I1205 03:38:27.908002  7359 AggregationFuzzer.cpp:576] [ROW ROW<g0:BOOLEAN,p1:ARRAY<SMALLINT>>: 3 elements, no nulls]
../../velox/exec/tests/utils/QueryAssertions.cpp:986: Failure
Value of: false
  Actual: false
Expected: true
Expected 3, got 3
1 extra rows, 1 missing rows
1 of extra rows:
	true | [75,269,374,482,941,1019,1677,1749,1895,2104,2179,2344,2556,2603,2682,2856,2900,2990,3460,4055,4062,4533,4972,5100,5525,5617,5868,6534,6664,6841,6945,7095,7358,7472,7851,7900,8337,8816,8901,9132,10104,11552,11584,11649,11851,11948,12821,12967,13206,13341,14199,15199,15875,15885,16479,16759,16847,16897,16996,17141,17274,17447,17802,18243,18250,18539,19199,19389,19638,20209,20432,20568,21387,21941,22052,22111,22378,22696,22828,23101,23323,23407,23426,23624,24018,24251,24318,24661,24786,24819,24905,24926,25637,26172,26232,26254,26329,26646,26980,27009,27405,27430,28246,28605,28681,28968,28981,29124,29165,29174,29727,29933,30098,30436,30449,30970,30979,31694,31984,32274,32412,32412,32472]

1 of missing rows:
	true | [75,269,374,482,941,1019,1677,1749,1895,2104,2179,2344,2556,2603,2682,2856,2900,2990,3460,4055,4062,4533,4972,5100,5525,5617,5868,6534,6664,6841,6945,7095,7358,7472,7851,7900,8337,8816,8901,9132,10104,11552,11584,11649,11851,11948,12821,12967,13206,13341,14199,15199,15875,15885,16479,16759,16847,16897,16996,17141,17274,17447,17802,18243,18250,18539,19199,19389,19638,20209,20432,20568,21387,21941,22052,22111,22378,22696,22828,23101,23323,23407,23426,23624,24018,24251,24318,24661,24786,24819,24905,24926,25637,26172,26232,26254,26329,26646,26980,27009,27405,27430,28246,28605,28681,28968,28981,29124,29165,29174,29727,29933,30098,30436,30449,30970,30979,31694,31984,32274,32412,32472]

Unexpected results
E1205 03:38:27.909212  7359 Exceptions.h:68] Line: ../../velox/exec/tests/AggregationFuzzer.cpp:786, Function:testPlan, Expression: assertEqualResults({expected.result}, {actual.result}) Logically equivalent plans produced different results, Source: RUNTIME, ErrorCode: INVALID_STATE
terminate called after throwing an instance of 'facebook::velox::VeloxRuntimeError'
  what():  Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: Logically equivalent plans produced different results
Retriable: False
Expression: assertEqualResults({expected.result}, {actual.result})
Function: testPlan
File: ../../velox/exec/tests/AggregationFuzzer.cpp
Line: 786
Stack trace:
# 0  _ZN8facebook5velox7process10StackTraceC1Ei
# 1  _ZN8facebook5velox14VeloxException5State4makeIZNS1_C4EPKcmS5_St17basic_string_viewIcSt11char_traitsIcEES9_S9_S9_bNS1_4TypeES9_EUlRT_E_EESt10shared_ptrIKS2_ESA_SB_
# 2  _ZN8facebook5velox14VeloxExceptionC1EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bNS1_4TypeES7_
# 3  _ZN8facebook5velox17VeloxRuntimeErrorC2EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bS7_
# 4  _ZN8facebook5velox6detail14veloxCheckFailINS0_17VeloxRuntimeErrorEPKcEEvRKNS1_18VeloxCheckFailArgsET0_
# 5  _ZN8facebook5velox4exec4test12_GLOBAL__N_117AggregationFuzzer8testPlanERKSt10shared_ptrIKNS0_4core8PlanNodeEEbbRKNS3_13ResultOrErrorE
# 6  _ZN8facebook5velox4exec4test12_GLOBAL__N_117AggregationFuzzer9testPlansERKSt6vectorISt10shared_ptrIKNS0_4core8PlanNodeEESaISA_EEbRKNS3_13ResultOrErrorE
# 7  _ZN8facebook5velox4exec4test12_GLOBAL__N_117AggregationFuzzer6verifyERKSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaISB_EESF_SF_RKS5_ISt10shared_ptrINS0_9RowVectorEESaISI_EEbSF_
# 8  _ZN8facebook5velox4exec4test12_GLOBAL__N_117AggregationFuzzer2goEv
# 9  _ZN8facebook5velox4exec4test15aggregateFuzzerESt13unordered_mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt6vectorISt10shared_ptrINS1_26AggregateFunctionSignatureEESaISD_EESt4hashIS9_ESt8equal_toIS9_ESaISt4pairIKS9_SF_EEEmRKS3_IS9_S9_SH_SJ_SaISK_ISL_S9_EEE
# 10 _ZN23AggregationFuzzerRunner3runERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEmRKSt13unordered_setIS5_St4hashIS5_ESt8equal_toIS5_ESaIS5_EERKSt13unordered_mapIS5_S5_SA_SC_SaISt4pairIS6_S5_EEE
# 11 main
# 12 __libc_start_main
# 13 _start

*** Aborted at 1670211508 (Unix time, try 'date -d @1670211508') ***
*** Signal 6 (SIGABRT) (0x1cbf) received by PID 7359 (pthread TID 0x7faec649d380) (linux TID 7359) (maybe from PID 7359, UID 0) (code: -6), stack trace: ***
(error retrieving stack trace)
```

**Comment by mbasmanova** [comment]
Looks like there is a duplicate map key: `...,32412,32412,...`

**Comment by mbasmanova** [comment]
This is probably fixed now. See https://github.com/facebookincubator/velox/issues/3402#issuecomment-1343351785

**Comment by mbasmanova** [comment]
Confirmed that this is fixed in #3291
